AD_query_tool.py

```diff
@@ -0,0 +1,99 @@
+
+import os
+import ldap
+import csv
+import pandas as pd
+import datetime
+import dateutil.relativedelta
+
+
+def active_directory_conn(ldap_server, ldap_login, ldap_pass):
+    conn = ldap.initialize(ldap_server)
+    conn.simple_bind_s(ldap_login, ldap_pass) 
+    return(conn)
+
+def search_active_directory(base_dn, search_object,  search_attr, connection):
+    results = connection.search_s(base_dn, ldap.SCOPE_SUBTREE, search_object , search_attr)
+    connection.unbind()
+    return(results[:-4]) # cutting out irrelevant data rows 
+
+def results_to_csv(path, active_directory_query):
+    with open(path, 'w') as csv_file:  
+        writer = csv.writer(csv_file)
+        header = ['CN', 'GroupsMembership', 'UserAccountControl', 'LastLogon', 'LastPasswordSet'] 
+        writer.writerow(header)
+        for i in active_directory_query:
+            cn = i[0]
+            member_of = i[0],i[1].get('memberOf')
+            user_account_control = int(i[1].get('userAccountControl')[0])
+            last_logon = int(i[1].get('lastLogon')[0])
+            pwd_last_set = int(i[1].get('pwdLastSet')[0])
+            writer.writerow([cn, member_of ,user_account_control, last_logon, pwd_last_set])
+
+def enabled_users(df):    
+    enabled_users = df.loc[df['UserAccountControl'] & 2 == 0]
+    return(enabled_users)
+
+def enabled_users_amount():    
+    df = pd.read_csv(csv_path)
+    enabled_users_amount = enabled_users(df)['CN'].count()
+    return(enabled_users_amount)
+
+def current_date_time():
+    current_ts = datetime.datetime.now()
+    current_date_time = current_ts.replace(microsecond=0)
+    return(current_date_time)
+
+def to_date_time(timestamp):
```

Comment:
I don't understand this function.
Maybe add some documentation here about what it's doing

Reply:
This function converts the 18-digit Active Directory timestamp to human readable datetime. I've added this comment to the code

Reply:
good, i would also make it clear in the name of the function, e.g "ldap_timestamp_to_datetime"